qmxgraph/html.py

```diff
@@ -1,22 +1,108 @@
-def create_image_tag_str(source, width, height):
+import attr
+
+
+asdict = attr.asdict
+
+
+@attr.s(repr=False, slots=True, hash=True)
+class _HtmlChildTypeValidator(object):
```

Comment:
Also I believe a better name would be `tuple_of` to follow with the other validator names:

```python
contents = attr.ib(validator=tuple_of(str, Image),
                   convert=tuple)
```